### ISSUE TITLE
Order detail displayed twice when order is selected in Order Dashboard

### ISSUE THREAD

**Issue by zenweasel** [reactioncommerce/reaction#2959]
### Expected behavior
Order detail should only show up once

### Actual Behavior
Shows up twice with inconsistent results
![order_details_duplicated](https://user-images.githubusercontent.com/72819/30891670-26b69ddc-a367-11e7-9a41-0ee00bf0695f.gif)

### Steps to Reproduce the Behavior
1. Login as admin and enable shipping/payment
1. Place an item in cart and checkout
1. Place a second order
1. Go to order dashboard
1. Select an order
1. Observe order is listed in details pane
1. Select 2nd order.
1. Observe that previous order and current order are both visible in the detail pane

### Versions
```
Node: 7.10.0
NPM: 5.3.0
Meteor Node: 4.8.4
Meteor NPM: 4.6.1
Reaction CLI: 0.12.0
Reaction: 1.5.0
Reaction branch: spencer-fix-2955-order-publication-userid (marketplace)
Docker: 17.06.2-ce
```

**Comment by joykare** [comment]
*Notes*
- It displays twice because of this line right here: https://github.com/reactioncommerce/reaction/blob/marketplace/imports/plugins/core/orders/client/templates/workflow/workflow.html#L4
- When you have an order and ship, two shipping items are added into the shipping array in the DB
- Next step: figure out why there are two shipping items for one order and if they are supposed to exist in the first place

**Comment by joykare** [comment]
- Observed this behavior on old orders I had. By behavior I mean:

<img width="1192" alt="screen shot 2017-10-04 at 22 15 03" src="https://user-images.githubusercontent.com/19430074/31195046-a69b6498-a951-11e7-9418-3275c0a7a82d.png">
Unable to replicate this on new orders and after `reaction reset`.

@zenweasel @kieha could you try reproducing this bug after a reaction reset/ just on new orders.

**Comment by kieha** [comment]
Every first order of every user seems to come with two shipping objects that have the same shopId. The subsequent orders all have one shipping object. This is after a `reaction reset`.

**Comment by impactmass** [comment]
To add also, for those two shipping objects mentioned, there's a success/error field in there i.e `shipping.shipmentQuotesQueryStatus.requestStatus`.  (cc @jm-wanja you mentioned this too somewhere, right?)

**Comment by joykare** [comment]
@impactmass seen it. Thanks for mentioning it.

*Update*
- Figuring out why this is happening

**Comment by zenweasel** [comment]
@joykare I see this issue all day with new orders and I'm constantly resetting. Let me try and drill down into some better reproduction steps.

**Comment by zenweasel** [comment]
I think the two shipping records is a bug introduced by me that I have fixed on #2847

**Comment by joykare** [comment]
@zenweasel cool cool, my findings so far, however, is that it happens only on the first order on a new user. All other subsequent orders by the same user are okay.

Reproduction steps for different findings are highly welcome.

**Comment by joykare** [comment]
@zenweasel just tested for this on your PR and its fixed (you could add this as a bug your PR resolves_unless you can reproduce it some other way)
- The shipping having two items for the same order was happening for the first order for every new user

**Comment by spencern** [comment]
Resolved by #2847
